sxs/waveforms/alignment.py

```diff
@@ -310,3 +297,226 @@ def align2d(wa, wb, t1, t2, n_brute_force_δt=None, n_brute_force_δϕ=5, includ
     idx = np.argmin(abs(np.array([optimum.cost for optimum in optimums])))
 
     return optimums[idx].cost, wa_primes[idx], optimums[idx]
+
+
+def _cost4d(δt_δSO3, args):
+    from .. import WaveformModes
+
```

Comment:
np.quaternion only exists if you import quaternion.  We really have to use quaternionic.  I'm not sure I'll even be able to maintain quaternion now that numpy 2.0 has broken everything.